src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSource.cs

```diff
@@ -29,5 +29,7 @@ public abstract partial class DiagnosticSource
         public virtual bool IsEnabled(string name, object? arg1, object? arg2 = null) { throw null; }
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("The type of object being written to DiagnosticSource cannot be discovered statically.")]
         public abstract void Write(string name, object? value);
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The type of object being written to DiagnosticSource cannot be discovered statically.")]
```

Comment:
I need to fix up this message. These APIs are still marked RUC, but we need to tell developers that properties on derived types, and recursive properties may be trimmed.